mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -479,10 +481,11 @@ static void printAttributions(OpAsmPrinter &p, StringRef keyword,
   if (values.empty())
     return;
 
-  p << ' ' << keyword << '(';
-  llvm::interleaveComma(
-      values, p, [&p](BlockArgument v) { p << v << " : " << v.getType(); });
-  p << ')';
+  auto printBlockArg = [](BlockArgument v) {
+    return llvm::formatv("{} : {}", v, v.getType());
```

Comment:
This code is technically more expensive as `llvm::formatv` allocates a temp string, but I don't think it matters for printers much.

Reply:
Ack